app/controllers/admin_controller.rb

```diff
@@ -168,6 +168,13 @@ def publish
           AdminMailer.notify_author_of_approval(@node, current_user).deliver_now
           AdminMailer.notify_moderators_of_approval(@node, current_user).deliver_now
           SubscriptionMailer.notify_node_creation(@node).deliver_now
+
+          if @node.tags.count > 0
+              @node.tags.each do |tag|
+                SubscriptionMailer.notify_tag_added(@node, tag, @node.author).deliver_now unless tag.subscriptions.empty?
```

Comment:
This looks good, but the only issue is that people may receive multiple emails now. I want to suggest you look at how we collect emails in this PR (unfinished!) and send a single notification command to only unique emails (I had to make the email text a little more generic to accomplish this) -- https://github.com/publiclab/plots2/pull/2742/files

What do you think?

Reply:
In our current system, we send a new mail for each tag addition like if a user is subscribed to three tags say `test1`,`test2`, `test3` , he will get 3 emails if a note is created with them. So, we need to develop a system where only  1 mail should be sent containing text such as `a note is generated with these 3 tags `. Am I getting it right?